tests/test_data_adapter.py

```diff
@@ -9,11 +9,71 @@
 import xarray as xr
 import os
 import hydromt
-from hydromt.data_adapter import DataAdapter, DataCatalog
+from hydromt.data_adapter import (
+    DataAdapter,
+    DataCatalog,
+    RasterDatasetAdapter,
+    _parse_data_dict,
+)
 
 TESTDATADIR = join(dirname(abspath(__file__)), "data")
 
 
+def test_parser():
+    # simple; abs path
+    root = "c:/root"
+    dd = {
+        "test": {
+            "data_type": "RasterDataset",
+            "path": "d:/path/to/data.tif",
+        }
+    }
+    dd_out = _parse_data_dict(dd, root=root)
+    assert isinstance(dd_out["test"], RasterDatasetAdapter)
+    assert dd_out["test"].path == abspath(dd["test"]["path"])
+    # rel path
+    dd = {
+        "test": {
+            "data_type": "RasterDataset",
+            "path": "path/to/data.tif",
+            "kwargs": {"fn": "test"},
+        },
+        "root": root,
+    }
+    dd_out = _parse_data_dict(dd)
+    assert dd_out["test"].path == abspath(join(root, dd["test"]["path"]))
+    # check if path in kwargs is also absolute
+    assert dd_out["test"].kwargs["fn"] == abspath(join(root, "test"))
+    # alias
+    dd = {
+        "test": {
+            "data_type": "RasterDataset",
+            "path": "path/to/data.tif",
+        },
+        "test1": {"alias": "test"},
+    }
+    dd_out = _parse_data_dict(dd, root=root)
+    assert dd_out["test"].path == dd_out["test1"].path
+    # placeholder
```

Comment:
@laurenebouaziz I added unit test for this method. Can you make sure that the part on placeholders makes sense to you?

Reply:
looks good to me!